vitrina/views.py

```diff
@@ -13,3 +17,58 @@ def home(request):
             'project': Project.public.count(),
         }
     })
+
+
+class HistoryView(PermissionRequiredMixin, TemplateView):
+    template_name = 'history.html'
+    model = None
+    detail_url_name = None
+    history_url_name = None
+
+    def has_permission(self):
+        obj_id = self.kwargs.get('pk')
+        obj = get_object_or_404(self.model, pk=obj_id)
+        return can_manage_history(obj, self.request.user)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        obj_id = self.kwargs.get('pk')
+        obj = get_object_or_404(self.model, pk=obj_id)
+        extra_context = {
+            "detail_url_name": self.get_detail_url_name(),
+            "history_url_name": self.get_history_url_name(),
+            "history": [{
+                'date': version.revision.date_created,
+                'user': version.revision.user,
+                'action': self.model.HISTORY_MESSAGES.get(version.revision.comment),
+            } for version in Version.objects.get_for_object(obj).order_by('-revision__date_created')],
+        }
+        context.update(extra_context)
+        return context
+
+    def get_detail_url_name(self):
+        return self.detail_url_name
+
+    def get_history_url_name(self):
+        return self.history_url_name
+
+
+class HistoryDetailView(DetailView):
```

Comment:
```suggestion
class HistoryMixin:
```

View classes are extended using mixins.

https://docs.djangoproject.com/en/3.2/topics/class-based-views/mixins/